src/util.js

```diff
@@ -276,6 +281,85 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+global.eventsToHandlers = global.eventsToHandlers ? global.eventsToHandlers :
+    {};
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
+// for debug methods
+function hookTo(prototypeObject, methodName, extension) {
+    var existingMethodImpl = prototypeObject[methodName];
+    var extensions = null;
+    if (typeof existingMethodImpl === "function") {
+        extensions = global.eventsToHandlers[methodName];
+        if (!extensions) {
+            extensions = global.eventsToHandlers[methodName]
+                = [];
+        }
+        if (extensions.indexOf(extension) >= 0) {
+            throw new Error("Trying to register same extension method twice");
```

Comment:
Because it's an error with the type, it's kind of a stretch though. What error type would you recommend? I guess since this is internal it's not a big deal.